Meta/ladybird.py

```diff
@@ -0,0 +1,295 @@
+        return build_preset_dir
+
+    _build_vcpkg()
+    _validate_cmake_version()
+
+    cmake_args.extend(build_env_cmake_args)
+    config_args = [
+        'cmake',
+        '--preset',
+        kwargs.get('preset'),
+        '-S',
+        lb_source_dir,
+        '-B',
+        build_preset_dir,
+    ]
+    config_args.extend(cmake_args)
+
+    try:
+        subprocess.check_call(config_args)
+    except subprocess.CalledProcessError as e:
+        _print_process_stderr(e, 'Unable to configure ladybird project')
+        sys.exit(1)
+
+    return build_preset_dir
+
+
+def _configure_skia_jemalloc():
+    import resource
+    page_size = resource.getpagesize()
+    gn = shutil.which('gn') or None
+    # https://github.com/LadybirdBrowser/ladybird/issues/261
+    if page_size != 4096 and gn is None:
+        print('GN not found! Please build GN from source and put it in $PATH', file=sys.stderr)
+        sys.exit(1)
+    pkg_config = shutil.which('pkg-config') or None
+    cmake_args = []
+    if pkg_config:
+        cmake_args.append(f'-DPKG_CONFIG_EXECUTABLE={pkg_config}')
+
+    user_vars_cmake_module = Path('Meta/CMake/vcpkg/user-variables.cmake')
+    user_vars_cmake_module.parent.mkdir(parents=True, exist_ok=True)
+
+    with open(user_vars_cmake_module, 'w') as f:
+        f.writelines([
+            f'set(PKGCONFIG {pkg_config})',
+            f'set(GN {gn})',
+            '',
+        ])
+
+    return cmake_args
+
+
+def _configure_build_env(**kwargs):
+    preset = kwargs.get('preset')
+    cc = kwargs.get('cc')
+    cxx = kwargs.get('cxx')
+    cmake_args = []
+    cmake_args.extend([
+        f'-DCMAKE_C_COMPILER={cc}',
+        f'-DCMAKE_CXX_COMPILER={cxx}',
+    ])
+    _ensure_ladybird_source_dir()
+    lb_source_dir = Path(os.environ.get('LADYBIRD_SOURCE_DIR'))
+
+    build_root_dir = lb_source_dir / 'Build'
+
+    known_presets = {
+        'default': build_root_dir / 'release',
+        'windows_ci_ninja': build_root_dir / 'release',
+        'windows_dev_ninja': build_root_dir / 'debug',
+        'windows_dev_msbuild': build_root_dir / 'debug',
+        'Debug': build_root_dir / 'debug',
+        'Sanitizer': build_root_dir / 'sanitizers',
+        'Distribution': build_root_dir / 'distribution',
+    }
+    if preset not in known_presets:
+        print(f'Unknown build preset "{preset}"', file=sys.stderr)
+        sys.exit(1)
+
+    build_preset_dir = known_presets.get(preset)
+    vcpkg_root = str(build_root_dir / 'vcpkg')
+    os.environ['VCPKG_ROOT'] = vcpkg_root
+    os.environ['PATH'] += os.pathsep + str(lb_source_dir.joinpath('Toolchain', 'Local', 'cmake', 'bin'))
+    os.environ['PATH'] += os.pathsep + str(vcpkg_root)
+    return lb_source_dir, build_preset_dir, cmake_args
+
+
+def _build_vcpkg():
+    sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'Toolchain'))
+    from BuildVcpkg import main as build_vcpkg
```

Comment:
a separate patch to actually rename that function to `build_vcpkg` function and call it from `__main__` in the original file would be in order.